macros/clone_schema.sql

```diff
@@ -0,0 +1,266 @@
+                    END IF;
+
+                END LOOP;
+            END IF;
+
+            --4. Add foreign key constraints.
+            FOR qry IN
+                SELECT 'ALTER TABLE ' || quote_ident(dest_schema) || '.' || quote_ident(rn.relname) 
+                                    || ' ADD CONSTRAINT ' || quote_ident(ct.conname) || ' ' || pg_get_constraintdef(ct.oid) || ';'
+                FROM pg_constraint ct
+                INNER JOIN pg_class rn 
+                ON rn.oid = ct.conrelid
+                WHERE connamespace = src_oid
+                    AND rn.relkind = 'r'
+                    AND ct.contype = 'f'
+                LOOP
+                EXECUTE qry;
+                END LOOP;
+
+            --5. Create views.
+            IF comment_tag = '' THEN
+
+                SELECT count(*) INTO count_rows 
+                FROM information_schema.views
+                WHERE table_schema = quote_ident(source_schema);
+
+                IF count_rows > 0 THEN
+                    FOR object IN
+                        SELECT table_name::text,
+                            view_definition 
+                        FROM information_schema.views
+                        WHERE table_schema = quote_ident(source_schema)
+                    LOOP
+                        buffer := dest_schema || '.' || quote_ident(object);
+                        SELECT view_definition INTO v_def
+                        FROM information_schema.views
+                        WHERE table_schema = quote_ident(source_schema)
+                            AND table_name = quote_ident(object);
+                        EXECUTE 'CREATE OR REPLACE VIEW ' || buffer || ' AS ' || v_def || ';' ;
+                    END LOOP;
+
+                    EXECUTE 'SELECT description FROM pg_catalog.pg_description WHERE objoid = ''' || quote_ident(source_schema) || '.' || quote_ident(object) || '''::regclass::oid' INTO description_value; 
+                    IF description_value IS NOT NULL THEN
+                            EXECUTE 'COMMENT ON VIEW ' || quote_ident(dest_schema) || '.' || quote_ident(object)  || ' IS ''' || description_value ||''';'; 
+                    END IF;
+                END IF;
+            END IF;
+
+            --6. Create functions.
+            FOR func_oid IN
+                SELECT oid
+                FROM pg_proc 
+                WHERE pronamespace = src_oid
+            LOOP      
+                SELECT pg_get_functiondef(func_oid) INTO qry;
+                SELECT replace(qry, source_schema, dest_schema) INTO dest_qry;
+                EXECUTE dest_qry;
+            END LOOP;
+
+            RETURN; 
+            END;
+            $BODY$
+            LANGUAGE plpgsql VOLATILE
+            COST 100;
+
+        {% endset %} 
+
+        {% do run_query(set_pg_function) %}
+
+        {% set sql %}
+            SELECT clone_schema('{{schema_one}}', '{{schema_two}}', '{{comment_tag}}', true);
```

Comment:
Here we are setting SQL-statement calling to  clone_schema Postgres function declared before.